stdlib/atomic.mli

```diff
@@ -75,12 +74,25 @@ end @@ portable
 
 (* CR tdelvecchio: Document. *)
 
+val make : 'a @ contended -> 'a t
+[@@alert unsafe]
+
+val make_contended : 'a @ contended -> 'a t
+[@@alert unsafe]
+
 val get : 'a t -> 'a
 [@@alert unsafe]
 
-val exchange : 'a t -> 'a -> 'a
+val set : 'a t -> 'a @ contended -> unit
+[@@alert unsafe]
+
+val exchange : 'a t -> 'a @ contended -> 'a
 [@@alert unsafe]
 
+val compare_and_set : 'a t -> 'a @ contended -> 'a @ contended -> bool
+[@@alert unsafe]
```

Comment:
Nit: can we remove the mode annotations from the unsafe ones? Stronger discouragement to use them in places where you actually care about modes.